sound/soc/intel/boards/sof_pcm512x.c

```diff
@@ -86,6 +88,59 @@ static int sof_hdmi_init(struct snd_soc_pcm_runtime *rtd)
 
 static int sof_pcm512x_codec_init(struct snd_soc_pcm_runtime *rtd)
 {
+	struct sof_card_private *ctx = snd_soc_card_get_drvdata(rtd->card);
+	struct device *dev = rtd->card->dev;
+	unsigned int sck;
+	int ret;
+
+	ctx->sclk = devm_clk_get(rtd->card->dev, "sclk");
+	if (IS_ERR(ctx->sclk)) {
+		dev_info(dev, "Could not get SCLK, will operate in SOC master mode\n");
+		goto skip_dacpro;
```

Comment:
`return 0;`?

Reply:
better have a single exit point when there is a problem with the clock

Reply:
I'm not sure what you mean by "a problem with the clock," and how this is different from other error cases, but it's just cosmetics anyway.